aim/web/ui/src/pages/Home/components/HomeLeft/TagsCard/useTagsCard.tsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import _ from 'lodash-es';
+
+import { IExperimentData } from 'modules/core/api/experimentsApi';
+import { IResourceState } from 'modules/core/utils/createResource';
+import { Checkbox } from '@material-ui/core';
+
+import { Icon, Text } from 'components/kit';
+
+import createTagsEngine from './TagsStore';
+
+function useTagsCard() {
+  const tableRef = React.useRef<any>(null);
+  const [selectedRows, setSelectedRows] = React.useState<string[]>([]);
+  const { current: tagsEngine } = React.useRef(createTagsEngine);
+  const tagsStore: IResourceState<IExperimentData> = tagsEngine.tagsState(
+    (state) => state,
+  );
+
+  React.useEffect(() => {
+    tagsEngine.fetchTags();
+    return () => {
+      tagsEngine.tagsState.destroy();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // memoized table data
+  const tableData = React.useMemo(() => {
+    if (tagsStore.data) {
+      return tagsStore.data.map(
+        ({ name, archived, run_count }: any, index: number) => {
+          return {
+            key: index,
```

Comment:
let's set key as combination of index and name

Reply:
or if the id of the tag is the name, it means the name is unique, we can use the name as key